src/main/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsWriter.java

```diff
@@ -228,6 +238,12 @@ private <T, C> void trainAndIndex(
             : NativeIndexWriter.getWriter(fieldInfo, segmentWriteState);
 
         knnVectorValues = vectorValuesRetriever.apply(vectorDataType, fieldInfo, VectorProcessingContext);
+
+        StopWatch stopWatch = new StopWatch();
+        stopWatch.start();
         indexOperation.buildAndWrite(writer, knnVectorValues);
+        long time_in_millis = stopWatch.totalTime().millis();
+        graphBuildTime.set(graphBuildTime.getValue() + time_in_millis);
+        log.warn("Graph build complete in " + time_in_millis + " ms for " + graphBuildTime.getName());
```

Comment:
this code will not log what was the time taken for refresh and what time was taken during merge as `graphBuildTime.getName()` will always give `total_time_in_millis` for both of them.

I tested the code via this:

```
public void testABC() {
        System.out.println(KNNGraphValue.MERGE_TOTAL_TIME_IN_MILLIS.getName());
        Assert.fail();
    }
```

Output:
```
[2024-09-03T02:51:53,983][INFO ][o.o.k.i.c.KNNCodecFactoryTests] [testABC] before test
total_time_in_millis
[2024-09-03T02:51:54,037][INFO ][o.o.k.i.c.KNNCodecFactoryTests] [testABC] after test
```

Reply:
Two separate enums are used to track times for refresh and merge. This could be the resason.